### ISSUE TITLE
Using LightFM for small dataset with 1 positive interaction for most rows

### ISSUE THREAD

**Issue by hugos0910** [lyst/lightfm#366]
Hello,

I've constructed an interaction matrix from my data, with the shape of (23609, 119).  The problem I am having is that for most of the rows there is only 1 positive interaction.  The distribution of the number of interaction is below:

* Number of rows with 1 interaction: 19572
* Number of rows with 2 interaction: 3184
* Number of rows with 3 interaction: 619
* Number of rows with 4 interaction: 179
* Number of rows with 5 interaction: 29
* Number of rows with 6 interaction: 14
* Number of rows with 7 interaction: 7
* Number of rows with 8 interaction: 5
* Number of rows with 9 interaction: 1

After the doing train_test_split, both of my training set and testing set are left with rows that are completely empty, with 0s only.  

I am encountering a lot of difficulties when it comes to examine my model with precision @ K.  The precision is extremely low (~0.01 to 0.03) because many rows have no known positive entry.

Any tips on how to work around this issue or choosing a different evaluation metric is appreciated, thanks!

**Comment by hugos0910** [comment]
Does that mean if ```user[0]``` has an interaction in the evaluation set but not the training set, this becomes a cold-start problem?

**Comment by maciejkula** [comment]
That's right. You may still be able to get reasonable predictions if you are using user features.

**Comment by SimonCW** [comment]
I’m closing this issue because it has been inactive for a long time. If you still encounter the problem, please open a new issue.

Thank you!
